src/components/ComponentList.js

```diff
@@ -104,25 +104,35 @@ export default class ComponentList extends React.Component {
     const { readOnly } = this.props
     const isSourceComponent = this.isSourceComponent(component)
     const scores = Definition.computeScores(definition)
+    const isDisabled = Definition.isDefinitionEmpty(definition)
     return (
       <div className="list-activity-area">
         {scores && <img className="list-buttons" src={getBadgeUrl(scores.tool, scores.effective)} alt="score" />}
         <ButtonGroup>
           {!isSourceComponent &&
-            !readOnly && (
-              <Button className="list-hybrid-button" onClick={this.addSourceForComponent.bind(this, component)}>
-                <i className="fas fa-plus" />
-                <span>&nbsp;Add source</span>
-              </Button>
+            !readOnly &&
+            this.renderButtonWithTip(
+              <Button onClick={this.addSourceForComponent.bind(this, component)}>
+                <i className="fas fa-code" />
+              </Button>,
+              'Add the definition for source that matches this package'
             )}
           {this.renderButtonWithTip(
-            <Button className="list-fa-button" onClick={this.inspectComponent.bind(this, currentComponent, definition)}>
+            <Button
```

Comment:
better here would be to only show the "add source" button if there is actually a sourcelocation in the given definition. That works for the empty case as well as improving the experience when there is a definition and it just happens to not have a source location.

Also means (I think) that you can get rid of the whole "disabled" concept.

Reply:
@jeffmcaffer i've updated the code based on your suggestion.

This is what is gonna happen if the definition has an empty sourceLocation:
![schermata 2018-09-28 alle 09 31 19](https://user-images.githubusercontent.com/7654979/46194300-4f712980-c301-11e8-9fa4-c9b5188b2a6e.png)

Reply:
@jeffmcaffer what should be the best way to check if a definition is empty?

Actually, a definition that hasn't any data, has a structure like this:
![schermata 2018-09-28 alle 15 30 51](https://user-images.githubusercontent.com/7654979/46211312-82ccac00-c333-11e8-93f7-d1a2e87e2602.png)

Given that, we checked for the existence of the field "sourceLocation".
That's why, from our perspective, the best way to check an empty definition was to check the sourceLocation field.